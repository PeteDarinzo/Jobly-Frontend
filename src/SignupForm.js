import React, { useState } from "react";
import { Button, Form, FormGroup, Label, Input, Alert } from 'reactstrap';
import "./Form.css";



const SignupForm = ({ register, error }) => {

  const initialState = {
    username: "",
    password: "",
    firstName: "",
    lastName: "",
    email: ""
  }

  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    e.preventDefault();
    const { name, value } = e.target;
    setFormData(formData => ({ ...formData, [name]: value }));
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    register(formData);
    setFormData(initialState);
  }

  return (
    <Form onSubmit={handleSubmit} className="Form shadow">
      <h1>Sign Up</h1>
      {error && <Alert color="danger">
        {error}
      </Alert>}
      <FormGroup>
        <Label for="username">Username</Label>
        <Input
          type="text"
          name="username"
          id="username"
          placeholder="username"
          onChange={handleChange}
          required
        />
      </FormGroup>
      <FormGroup>
        <Label for="examplePassword">Password</Label>
        <Input
          type="password"
          name="password"
          id="examplePassword"
          placeholder="password"
          onChange={handleChange}
          required
        />
      </FormGroup>
      <FormGroup>
        <Label for="firstName">First Name</Label>
        <Input
          type="text"
          name="firstName"
          id="firstName"
          placeholder="first name"
          onChange={handleChange}
          required
        />
      </FormGroup>
      <FormGroup>
        <Label for="lastName">Last Name</Label>
        <Input
          type="text"
          name="lastName"
          id="lastName"
          placeholder="last name"
          onChange={handleChange}
          required
        />
      </FormGroup>
      <FormGroup>
        <Label for="exampleEmail">Email</Label>
        <Input
          type="email"
          name="email"
          id="exampleEmail"
          placeholder="email"
          onChange={handleChange}
          required
        />
      </FormGroup>
      <Button color="primary" className="btn-lg">Submit</Button>
    </Form>
  );
}

export default SignupForm;