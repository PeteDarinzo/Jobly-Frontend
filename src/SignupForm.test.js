import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom"
import SignupForm from "./SignupForm";

it("renders signup form without crashing", () => {
  render(
    <MemoryRouter>
      <SignupForm register={() => { }} error="" />
    </MemoryRouter>
  );
});

it("matches the home snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <SignupForm register={() => { }} error="" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});