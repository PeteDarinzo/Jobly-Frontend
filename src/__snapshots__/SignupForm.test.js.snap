// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the home snapshot 1`] = `
<DocumentFragment>
  <form
    class="Form shadow"
  >
    <h1>
      Sign Up
    </h1>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="username"
      >
        Username
      </label>
      <input
        class="form-control"
        id="username"
        name="username"
        placeholder="username"
        required=""
        type="text"
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="examplePassword"
      >
        Password
      </label>
      <input
        class="form-control"
        id="examplePassword"
        name="password"
        placeholder="password"
        required=""
        type="password"
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="firstName"
      >
        First Name
      </label>
      <input
        class="form-control"
        id="firstName"
        name="firstName"
        placeholder="first name"
        required=""
        type="text"
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="lastName"
      >
        Last Name
      </label>
      <input
        class="form-control"
        id="lastName"
        name="lastName"
        placeholder="last name"
        required=""
        type="text"
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="exampleEmail"
      >
        Email
      </label>
      <input
        class="form-control"
        id="exampleEmail"
        name="email"
        placeholder="email"
        required=""
        type="email"
      />
    </div>
    <button
      class="btn-lg btn btn-primary"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
